components/script/dom/htmlinputelement.rs

```diff
@@ -868,13 +886,14 @@ impl HTMLInputElement {
         let cx = GlobalScope::get_cx();
         let _ac = enter_realm(self);
         rooted!(in(*cx) let mut pattern = ptr::null_mut::<JSObject>());
+        let can_gc = CanGc::note();
 
-        if compile_pattern(cx, &pattern_str, pattern.handle_mut()) {
+        if compile_pattern(cx, &pattern_str, pattern.handle_mut(), can_gc) {
             if self.Multiple() && self.does_multiple_apply() {
                 !split_commas(value)
-                    .all(|s| matches_js_regex(cx, pattern.handle(), s).unwrap_or(true))
+                    .all(|s| matches_js_regex(cx, pattern.handle(), s, can_gc).unwrap_or(true))
             } else {
-                !matches_js_regex(cx, pattern.handle(), value).unwrap_or(true)
+                !matches_js_regex(cx, pattern.handle(), value, CanGc::note()).unwrap_or(true)
```

Comment:
Note to self: file followup to propagate CanGc to suffers_from_pattern_mismatch and callers.